pkg/urfav_overrides/flags.go

```diff
@@ -24,9 +24,13 @@ type Flags struct {
 // to use this in a command,
 // This package interacts with os.Args directly
 //
-// allFlags := cfflags.New("name",GlobalFlagsList, c)
+// allFlags := cfflags.New("name",GlobalFlagsList, c, 1)
 // allFlags.String("region")
-func New(name string, flags []cli.Flag, c *cli.Context) (*Flags, error) {
+// When using granted global flags for a subcommand. Eg. granted registry add {URL:} -c ref. you will need to need to specify a command depth
+// for it to work correctly.
+// The depth is used to filter out the subcommands when grabbing out the flags passed in the body of the cli request
+// err = set.Parse(ag) fails to populate the flag value when args includes subcommands so we filter them out
+func New(name string, flags []cli.Flag, c *cli.Context, depth int) (*Flags, error) {
```

Comment:
The purpose of this package is to allow flags to be provided in any order for the `assume` command.

e.g.

```
assume -c my-profile
assume my-profile -c
```
The built in `urfav/cli` argument parser doesn't support this so we needed to extend it ourselves.

For the `granted registry` commands though we don't need to support flags being placed in any order.
For the `--ref` flag as an example, the only accepted order should be:

```
granted registry add --ref dev <registry url>
```

This behaviour is how `urfav/cli` works already, so we don't need to use this package at all.

Looking at the changes in this PR it looks like you have attempted to extend the `urfav_overrides` package to get this behaviour, but it is already built in.

I would propose that we revert the changes on this package (i.e. remove the `depth` argument), and then just use built in argument parsing, i.e.

```go
var MyCommand = cli.Command{
	Name: "command",
	Flags: []cli.Flag{
		&cli.StringFlag{Name: "ref"},
	},
	Action: func(c *cli.Context) error {
		ref := c.String("ref")
		// do stuff with ref

		return nil
	},
}
```

Reply:
@meyerjrr Tagging you here as this was implemented by you.

Reply:
@Eddie023 just revert these changes directly rather than waiting for Jack's okay!